src/logger/BaseLogger.go

```diff
@@ -79,39 +56,58 @@ func NewBaseLogger(_agent string, _url string, _enabled bool, _queue []string) *
 	return constructedBaseLogger
 }
 
-func (obj BaseLogger) Enable() {
-	obj.enabled = true
+func (logger BaseLogger) Enable() {
+	logger.enabled = true
 }
 
-func (obj BaseLogger) Disable() {
-	obj.enabled = false
+func (logger BaseLogger) Disable() {
+	logger.enabled = false
 }
 
 /**
  * Submits JSON message to intended destination.
  */
-func (obj BaseLogger) Submit(msg string) {
+func (logger BaseLogger) Submit(msg string) {
 	//woah congrats you submitted the message
-	//TODO: implement submit func
-	if msg == "" || obj.SkipSubmission() || !obj.Enabled() {
+	if msg == "" || logger.SkipSubmission() || !logger.Enabled() {
 		//do nothing
-	} else if obj.queue != nil {
-		queue.add(msg)
-		obj.submitSuccesses.increment()
+	} else if logger.queue != nil {
+		logger.queue = append(logger.queue, msg)
+		atomic.AddInt64(&logger.submitSuccesses, 1)
 	} else {
-		// HttpURLConnection url_connection = (HttpURLConnection) this.url_parsed.openConnection();
-		// url_connection.setConnectTimeout(5000);
-		// url_connection.setReadTimeout(1000);
-		// url_connection.setRequestMethod("POST");
-		// url_connection.setRequestProperty("Content-Type", "application/json; charset=UTF-8");
-		// url_connection.setRequestProperty("User-Agent", "Resurface/" + version + " (" + agent + ")");
-		// url_connection.setDoOutput(true);
-		if obj.SkipCompression() {
-			url_connection.setRequestProperty("Content-Encoding", "deflated")
-		} else {
+		// not 100% sure this works (needs testing)
+		submitRequest := http.NewRequest("POST", logger.url, bytes.NewBuffer([]byte(msg)))
```

Comment:
I'm guessing this is for building the Request Header without setting all the of different fields that come with it.